lib/io.js

```diff
@@ -535,9 +536,27 @@
           var cons = f.cat(router.filter(connected), newConnects);
           var newRouter = new ConsistentHashRouter(
             logger, topology.id, cons, topology.segments, router.getClusterName());
-          emitter.emit('router', newRouter);
-        })
-        .then(failoverListeners(transport, missing));
+          onRouter(newRouter);
+          return newConnects;
+        }).then(function (conns) {
+            return Promise.all(_.map(conns, function (conn) {
+              return conn.connect().then(function() {
```

Comment:
I'd rework this because you always want to ping before attempting authentication, since, starting with Hot Rod 3.0, ping provides version handshaking capabilities
I'd therefore go with something like connect > ping > if (auth) auth

Reply:
we talked about. the fact that ping auth is ping in a way as well, so in this version if not auth, then ping. 
can we tackle this in ping 3.0?